buf/registry/plugin/v1beta1/commit.proto

```diff
@@ -0,0 +1,84 @@
+// Copyright 2023-2024 Buf Technologies, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+syntax = "proto3";
+
+package buf.registry.plugin.v1beta1;
+
+import "buf/registry/plugin/v1beta1/digest.proto";
+import "buf/registry/priv/extension/v1beta1/extension.proto";
+import "buf/validate/validate.proto";
+import "google/protobuf/timestamp.proto";
+
+option go_package = "buf.build/gen/go/bufbuild/registry/protocolbuffers/go/buf/registry/plugin/v1beta1";
+
+// A commit on a specific Plugin.
+//
+// Commits are immutable.
+//
+// Many Commits may be associated with one Digest.
+//
+// Not that the Digest returned on a Commit depends on the requested DigestType in the RPC that
+// returned the Commit.
+message Commit {
+  option (buf.registry.priv.extension.v1beta1.message).response_only = true;
+
+  // The id of the Commit.
+  string id = 1 [
+    (buf.validate.field).required = true,
+    (buf.validate.field).string.tuuid = true
+  ];
+  // The time the Commit was pushed to the BSR.
+  //
+  // Commits are immutable, so there is no corresponding update_time.
+  google.protobuf.Timestamp create_time = 2 [(buf.validate.field).required = true];
+  // The id of the Organization that owns the Plugin that the Commit is associated with.
+  string owner_id = 3 [
+    (buf.validate.field).required = true,
+    (buf.validate.field).string.tuuid = true
+  ];
+  // The id of the Plugin that the Commit is associated with.
+  string plugin_id = 4 [
+    (buf.validate.field).required = true,
+    (buf.validate.field).string.tuuid = true
+  ];
+  // The Digest of the Commit's contents.
+  Digest digest = 5 [(buf.validate.field).required = true];
+  // The id of the User that created this Commit on the BSR.
+  //
+  // May be empty if the User is no longer available.
+  string created_by_user_id = 6 [
```

Comment:
Why or why not is `source_control_url` not a field here, when it is for the module commit?

Reply:
What is the answer to why or why not this field was not present in this PR already? This determines how you think about this solution.

Reply:
I excluded it because it (initially) made less sense for plugins. For modules, you commit proto files, license, readme to vcs. There's a direct link between the committed files, a commit and the resulting contents pushed up to the BSR.

The GitHub action pushes each vcs commit (regardless of proto changes) to the BSR.

For plugins, you're typically not building and committing the compiled wasm module into vcs. And you're very unlikely to be publishing for every vcs commit, instead, you'd be doing a bunch of work on the plugin and when it's ready tag it and release. The model felt slightly different here.

I agree we should add this field to the `Commit` response object though. If a plugin from the BSR can point to the corresponding source code from which it was built then it's a net win.

Reply:
It seems to me that there's no reason to differentiate here - I would add `buf plugin push --source-control-url`